crux.el

```diff
@@ -203,14 +203,30 @@ With a prefix ARG open line above the current line."
 Passes ARG to command `kill-whole-line' when provided."
   (interactive "p")
   (kill-whole-line arg)
-  (back-to-indentation))
+  (move-to-smart-line-start))
 
 (defun crux-kill-line-backwards ()
   "Kill line backwards and adjust the indentation."
   (interactive)
   (kill-line 0)
   (indent-according-to-mode))
 
+
+(defvar crux-smart-line-start-regex "^[[:space:]]*"
+  "Regex to match pre-line text.
+Matches whitespace in nearly all major modes.  Matches the terminal prompts
+ in 'ansi-term' and 'eshell.")
+(add-hook 'term-mode-hook   (lambda () (setq crux-smart-line-start-regex "^[^#$%>\n]*[#$%>] " )))
```

Comment:
Seems to me it'd be better if those were separate vars/constants and you added a check inside the function for the current mode. Btw, keep in mind that for your variable solution to work this variable has to be `buffer-local`, not global like it is now.